### PR TITLE
Add ability to mask card number

### PR DESCRIPTION
**Pull request by hernan**
Related to #179 I implemented masking the card number on user input, added as a filter for numberInput handler.

### CODE REVIEW

**Comment by jessepollak** [comment]
Thanks for your PR and sorry for the long delay — just catching up! I'm going to leave a few comments, but I'm excited to merge this feature in!

File: src/coffee/card.coffee [hunk 1]
```diff
@@ -196,6 +201,19 @@ class Card
     QJ.toggleClass el, @options.classes.valid, test
     QJ.toggleClass el, @options.classes.invalid, !test
 
+  maskCardNumber: (val, el, out) =>
+    mask = @options.masks.cardNumber
+    cardType = payment.fns.cardType(val)
+    numbers = val.split(' ')
+
+    if (cardType == 'amex' && (numbers.length == 3)) || numbers.length >= 4
```
Comment (jessepollak) [inline src/coffee/card.coffee]: Can you explain why we have to add this exception for amex? I'm a little bit wary of adding card type specific code here and wonder whether we can come up with a general solution.

**Comment by Soullivaneuh** [comment]
Maybe the same can be done for the CVV?

File: src/coffee/card.coffee (hunk 1 above)
Comment (hernan) [inline src/coffee/card.coffee]: Hi Jesse
The amex check isn't really necessary as I dont know much about card numbers and how many groups numbers can exists added amex check to make sure that my code will not break with other card type.
I made a quick check removing the cardType check and works without problems, but again I dont know all the details of card numbers.

**Comment by caiquecastro** [comment]
Haven't this PR been merged yet?

File: README.md [hunk 2]
```diff
@@ -183,7 +188,7 @@ $('form').card({
 ## Using with other javascript libraries
 
 Card has wrappers that make it easy to use with other javascript libraries:
-
```
Comment (Soullivaneuh) [inline README.md]: This should be done on a separated commit.

**Comment by BallisticPain** [comment]
@hernan Are you intending to remove the AMEX specific check since you said it works fine without it in there?

**Comment by hernan** [comment]
@BallisticPain adjusted the fix, please review.

**Comment by BallisticPain** [comment]
@hernan, please fix the latest conflict. I'll review right after. Sorry I missed your last comment to review.

**Comment by hernan** [comment]
Hi @BallisticPain is older than current master and don't see conflicts, I'm not sure how to proceed in this case to provide a correct commit, if you could guide me would be grat

**Comment by Paschoali** [comment]
@hernan Hi! Sorry if I am wrong, but you can remove all your changes on README.md file, backing it to original. Wait, are you working on branch master? Why? Anyway, if you back README.md to original (you can copy the README.md content from the server and paste on your file), update your branch with master branch from the server, later make your changes on README.md again and commit again, your branch will be synchronized with server and the approval of the PR will be easier.

**Comment by BallisticPain** [comment]
@hernan 
I would imagine you have this repository set as an upstream remote, is that correct?
`git remote -v` will give you the answer to the above.

When I have merge conflicts, I usually attempt to rebase the upstream branch onto mine. Below are the steps I would usually use to perform a rebase. Be sure you're on the master branch.

1. `git fetch upstream` - Grabs remote code from the upstream.
    * If you don't have an upstream... `git remote add upstream https://github.com/jessepollak/card.git`
2. `git rebase upstream/master` - Adds upstream commits before your commits.
3. Follow the instructions to fix any conflicts (should just be checking the conflicts in the README.md). Then run `git rebase --continue`
4. At this point your master should be up to date with upstream/master, but you will need to force the push up to your remote master `origin` via `git push --force origin master`
5. That will update this PR with your "fixed" branch.

Thanks for your continued responses and diligence!
Jarvis

**Comment by hernan** [comment]
@BallisticPain your guide worked without a glitch! thank you very much, to patch should be fine now, please review.

**Pull request merged**